### ISSUE TITLE
Add ability to directly invoke controller or app without saying connect UUID

### ISSUE THREAD

**Issue by timburke** [iotile/coretools#511]
Right now, if you just want to connect to a single device a single time and run an `app`, you need the following command line:

```
> iotile hw --port=PORT connect UUID app --name=NAME
```

This make sense from a code structure point of view since the `connect` method is on the `HardwareManager` object and the `app` method needs to be connected to a device already in order to query information about it and match it with the correct app.

However, it makes the command line very long.  A potential usability addition would be to support the following:

```
> iotile hw controller --device UUID
> iotile hw app --device UUID --name NAME
> iotile hw get 11 --device UUID
```

Basically, `HardwareManager` needs to know if it currently connected to a device and have an optional `device` keyword argument in the `get`, `controller` and `app` methods that, if the HardwareManager is not already connected, will just call `connect UUID` before proceeding.

This will reduce the amount of typing required for the common case of one-off device connections without impacting any other behavior.
